server/sonar-server/src/main/java/org/sonar/server/issue/ws/SetTypeAction.java

```diff
@@ -116,12 +122,11 @@ private SearchResponseData setType(DbSession session, String issueKey, RuleType
     IssueChangeContext context = IssueChangeContext.createUser(new Date(system2.now()), userSession.getLogin());
     if (issueFieldsSetter.setType(issue, ruleType, context)) {
       SearchResponseData searchResponseData = issueUpdater.saveIssueAndPreloadSearchResponseData(session, issue, context, null);
-      issueChangeTrigger.onChange(
-        new IssueChangeTrigger.IssueChangeData(
-          searchResponseData.getIssues().stream().map(IssueDto::toDefaultIssue).collect(MoreCollectors.toList(searchResponseData.getIssues().size())),
-          copyOf(searchResponseData.getComponents())),
-        new IssueChangeTrigger.IssueChange(ruleType),
-        context);
+      org.sonar.server.qualitygate.changeevent.QGChangeEventFactory.IssueChangeData issueChangeData = new QGChangeEventFactory.IssueChangeData(
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this fully qualified name with "QGChangeEventFactory" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)